beacon-chain/core/signing/signing_root.go

```diff
@@ -79,6 +79,27 @@ func ComputeDomainAndSign(st state.ReadOnlyBeaconState, epoch primitives.Epoch,
 	return key.Sign(sr[:]).Marshal(), nil
 }
 
+// ComputeDomainAndSignWithoutState offers the same functionalit as ComputeDomainAndSign without the need to provide a BeaconState.
+// This is particularly helpful for signing values in tests.
+func ComputeDomainAndSignWithoutState(fork *ethpb.Fork, epoch primitives.Epoch, domain [4]byte, vr []byte, obj fssz.HashRoot, key bls.SecretKey) ([]byte, error) {
```

Comment:
This makes tests much lighter, enabling us to generate signatures for blocks without generating a full state with correctly populated deterministic validators etc etc.

Reply:
Can't we use this internally in `ComputeDomainAndSign` ? Given that we only need the state to retrieve the current fork in the function. That we would still only have 1 implementation of signing a ssz object.

Reply:
Sure thing, I'll do that.

Reply:
https://github.com/prysmaticlabs/prysm/pull/13245/commits/944e6e38beceffefab4e7b759b73bd2c730a366f